### PR TITLE
feat : Adding support for AVG_RATE function in baseline rule

### PR DESCRIPTION
**Pull request by sarthak77**
**Description** 
Adding support for `avgrate` function alongside `sum` and `avg` function for aggregating metrics. 
Ref ticket : https://github.com/hypertrace/hypertrace-alert-engine/issues/39

**Test**
Wrote a unit test which fires a query to pinot and is working fine.

### CODE REVIEW

File: metric-anomaly-detector/src/test/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricAnomalyDetectorTest.java [hunk 1]
```diff
@@ -30,7 +31,7 @@
 
   @Test
   @Disabled
-  void testRuleEvaluation() throws URISyntaxException, IOException {
+  void staticTestRuleEvaluation() throws URISyntaxException, IOException {
```
Comment (rish691) [inline metric-anomaly-detector/src/test/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricAnomalyDetectorTest.java:34]: nit: testStaticRuleEvaluation

File: metric-anomaly-detector/src/test/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricAnomalyDetectorTest.java [hunk 2]
```diff
@@ -72,6 +74,77 @@ void testRuleEvaluation() throws URISyntaxException, IOException {
...
+     * start_time_millis >= ? AND start_time_millis < ? ) AND service_name = 'customer' ) GROUP BY
+     * dateTimeConvert(start_time_millis,'1:MILLISECONDS:EPOCH','1:MILLISECONDS:EPOCH','15:SECONDS')
+     * limit 10000
+     */
+    metricAnomalyDetector.process(alertTaskBuilder.build());
+  }
+
+  @Test
+  @Disabled
+  void dynamicTestRuleEvaluation() throws URISyntaxException, IOException {
```
Comment (rish691) [inline metric-anomaly-detector/src/test/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricAnomalyDetectorTest.java:108]: nit: testDynamicRuleEvaluation

File: metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/evaluator/BaselineRuleEvaluator.java [hunk 3]
```diff
@@ -74,6 +77,17 @@ public BaselineRuleEvaluator(QueryRequestHandler queryRequestHandler) {
                 && row.getColumn(1).getValueType()
                     == org.hypertrace.core.query.service.api.ValueType.STRING) {
               double value = Double.parseDouble(row.getColumn(1).getString());
+              if (metricAnomalyEventCondition.getMetricSelection().getMetricAggregationFunction()
```
Comment (rish691) [inline metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/evaluator/BaselineRuleEvaluator.java:80]: The same changes are needed in StaticRuleEvaluator as well?

File: metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/evaluator/BaselineRuleEvaluator.java [hunk 4]
```diff
@@ -196,4 +210,10 @@ private static Baseline getBaseline(double[] metricValueArray) {
     LOGGER.debug("Notification Event {}", notificationEvent);
     return Optional.of(notificationEvent);
   }
+
+  private double convertValue(long startTime, long endTime, double originalValue, String period) {
```
Comment (rish691) [inline metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/evaluator/BaselineRuleEvaluator.java:214]: method name is too generic, rename please

**Comment by rish691** [comment]
Build is failing

**Review by rish691 — APPROVED** [review]

**Review by kotharironak — COMMENTED** [review]
lgtm

**Pull request merged**